esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2298,6 +2266,70 @@ bool CFileSprayEx::onSprayVariable(IEspContext &context, IEspSprayVariable &req,
     return true;
 }
 
+void CFileSprayEx::checkDZScopeAccessAndSetSpraySourceDFUFileSpec(IEspContext &context, const char *srcPlane, const char *srcHost,
+    const char *srcFile, MemoryBuffer &srcXML, IDFUfileSpec *srcDFUfileSpec)
+{
+    if(srcXML.length() == 0)
+    {
+        //Both srcPlane and srcHost are validated in readAndCheckSpraySourceReq().
+        StringBuffer fnamebuf(srcFile);
+        fnamebuf.trim();
+        StringArray files;
+        files.appendList(fnamebuf, ","); // handles comma separated files
+        ForEachItemIn(i, files)
+        {
+            const char *file = files.item(i);
+            if (isEmptyString(file))
+                continue;
+
+            //Based on the tests, the dfuserver only supports the wildcard inside the file name, like '/path/f*'.
+            //The dfuserver throws an error if the wildcard is inside the path, like /p*ath/file.
+            StringBuffer dirPath, dirTail;
+            splitFilename(file, &dirPath, &dirPath, &dirTail, &dirTail);
+
+            const char* s = containsWildcard(dirTail) ? dirPath : file;
+            SecAccessFlags permission = getDropZoneScopePermissions(context, srcPlane, s, srcHost);
+            if (permission < SecAccess_Read)
+                throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Access DropZone Scope %s %s not allowed for user %s (permission:%s). Read Access Required.",
+                    srcPlane, s, context.queryUserId(), getSecAccessFlagName(permission));
+        }
+
+        SocketEndpoint ep(srcHost);
+        if (ep.isNull())
+            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Cannot resolve source network IP from %s.", srcHost);
+
+        RemoteMultiFilename rmfn;
+        rmfn.setEp(ep);
+        rmfn.append(fnamebuf.str());
+        srcDFUfileSpec->setMultiFilename(rmfn);
+    }
+    else
+    {
+        srcDFUfileSpec->setFromXML((const char*)srcXML.toByteArray());
```

Comment:
And, if we're not sure that this service is ever used like this, then we should at least comment that this may be deprecated/legacy.

Reply:
Also, in various places, it seems it just ignores the srcxml?
e.g. in readAndCheckSpraySourceReq, where readAndCheckSpraySourceReq is passed the request xml if it existed, but  if it has any content, the functionality is skipped..

Maybe, it's because after each call to readAndCheckSpraySourceReq, it calls checkDZScopeAccessAndSetSpraySourceDFUFileSpec, which does use/validate srcxml.
But it is unintuitive the way it is handling this.

I am not sure any of this srcxml handling should be supported.
Please open another JIRA (and link it to this one for reference) and find out whether this is really completely legacy, and if it is, we can remove it and make the code a lot clearer in this area.